src/main/react/app/routes/builder/context/node-context.tsx

```diff
@@ -0,0 +1,110 @@
+import useNodeContextStore from '~/stores/node-context-store'
+import { useEffect, useRef, useState } from 'react'
+import useFlowStore from '~/stores/flow-store'
+
+export default function NodeContext({
+  nodeId,
+  setShowNodeContext,
+}: Readonly<{
+  setShowNodeContext: (b: boolean) => void
+  nodeId: number
+}>) {
+  const attributes = useNodeContextStore((state) => state.attributes)
+  const inputReferences = useRef<Record<number, HTMLInputElement | null>>({})
+  const { setAttributes, setNodeName } = useFlowStore((state) => state)
+  const [canSave, setCanSave] = useState(false)
+
+  const validateForm = () => {
+    if (!attributes) return false
+    const allValid = attributes.every((attribute: any, index: number) => {
+      if (attribute.mandatory) {
+        const value = inputReferences.current[index]?.value?.trim()
+        return !!value
+      }
+      return true
+    })
+    setCanSave(allValid)
+  }
+
+  useEffect(() => {
+    if (!attributes) {
+      setCanSave(true)
+      return
+    }
+    validateForm()
+  }, [attributes])
+
+  const handleSave = () => {
+    const filledAttributes = Object.entries(inputReferences.current)
+      .map(([indexString, input]) => {
+        const index = Number(indexString)
```

Comment:
We're more a fan of this casting method (keeping it a primitive):
```suggestion
        const index = +indexString
```